### PR TITLE
[PotentialFlow] Preparing solver for shape optimization

### PR DESCRIPTION
**Pull request by marcnunezc**
**Description**
This prepares the solver to add a potential_flow_response script to perform optimization. The main change is to switch the ExecuteInitialize() of the wake to ExecuteInitializeSolutionStep().

Also a utility is added to compute the area given the container, to be used to compute the perimeter. 

**Changelog**
Please summarize the changes in one list to generate the changelog:
E.g.
- Change wake execution command.
- Add CalculateArea() utility

### CODE REVIEW

**Comment by philbucher** [comment]
This might have an effect on the CoSimApp tests that use potentialflow
I can check tmr

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def __init__(self, Model, settings):
             for node in cond.GetNodes():
                 node.Set(KratosMultiphysics.SOLID)
 
-    def ExecuteInitialize(self):
+    def ExecuteInitializeSolutionStep(self):
```
Comment (inigolcanalejo) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: Could you add a flag to decide whether to define the wake in each step or only once at the beginning? As you know, for some compressible and transonic cases we need to do load stepping to achieve convergence but it is not necessary to compute the wake in each step but rather only once at the beginning. In 2D defining the wake doesn't take much time but in 3D it is more expensive.

**Review by inigolcanalejo — CHANGES_REQUESTED** [review]
Really cool! Just a minor comment.

**Comment by inigolcanalejo** [comment]
> This might have an effect on the CoSimApp tests that use potentialflow
> I can check tmr

@philbucher I don't think this should affect the tests in the CoSimApp. As far as I remember to do FSI we need to define the wake not only in each time step but also at the beginning of each FSI iteration.

**Comment by philbucher** [comment]
> 
> 
> > This might have an effect on the CoSimApp tests that use potentialflow
> > I can check tmr
> 
> @philbucher I don't think this should affect the tests in the CoSimApp. As far as I remember to do FSI we need to define the wake not only in each time step but also at the beginning of each FSI iteration.

I would still run the tests to be sure :)

**Comment by inigolcanalejo** [comment]
> I would still run the tests to be sure :)

Aren't they anyway ran automatically in each PR?

**Comment by philbucher** [comment]
> 
> 
> > I would still run the tests to be sure :)
> 
> Aren't they anyway ran automatically in each PR?

only the small and fast ones

unfortunately coupled simulations take quite a while

**Comment by marcnunezc** [comment]
@philbucher , following @inigolcanalejo suggestion I have added a flag to control this. Now the default behaviour is the same as before, so for optimization cases I can activate the wake recomputation and that's it. I will remove the "Behaviour change" label from this PR.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py (hunk 1 above)
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: True, I actually like much more the option to activate/deactivate this behaviour. Thanks for the hint. Also, we should add the multi-step option in the apply_far_field_process for the transonic cases.

**Review by inigolcanalejo — APPROVED** [review]
Perfect, thanks for the effort!

**Comment by inigolcanalejo** [comment]
> > Aren't they anyway ran automatically in each PR?
> 
> only the small and fast ones
> 
> unfortunately coupled simulations take quite a while

Not with potential flow :wink:

**Pull request merged**